File: python/collatz.py
#usage: python3 collatz.py [number] [iterations]

import sys

if len(sys.argv) >= 3:
	n = int(sys.argv[1])

	for counter in range(int(sys.argv[2])):
		print(n)
		if n % 2 == 0:
			n = n / 2
		else:
			n = 3 * n + 1
	print(n)
else:
	print("usage: python3 collatz.py [number] [iterations]")

File: numeric_algos/collatz.cpp
#include <iostream>

using namespace std;

int collatz_tail(int number, int iterations){
    cout << number << endl;
    if (iterations == 0){
        return number;
    }
    if (number % 2 == 0){
        return collatz_tail(number / 2, iterations - 1);
    }
    return collatz_tail(3 * number + 1, iterations - 1);
}

int main(){
    cout << collatz_tail(12, 12) << endl;

}
